src/input/input.cpp

```diff
@@ -1493,6 +1499,13 @@ namespace Mist{
   }
 
   bool Input::readExistingHeader(){
+    const char *origTarget = getenv("MST_ORIG_TARGET");
+    if (!origTarget) {
```

Comment:
`MST_ORIG_TARGET` can be unset for many reasons. It only gets set by Mist when an Input gets started through the push system, so this might cause the header to be regenerated even if it is available for some situations. I would propose we take `config->getString("input")` by default and override it when `MST_ORIG_TARGET` is set instead